CDCgov/prime-reportstream
# React Network Layer: Hooks and APIs

This PR creates a new networking layer mvp for our React application by way of api classes (`Api.ts`, `History.ts` as examples), and the `useNetwork` hook. Additionally, this PR includes test setup, using `msw`, for positive and negative response mocking. 

Test Steps:
1. Pull branch
2. Run `yarn test` to ensure unit tests pass
3. Read the included `rs-react-networking.md` playbook
4. Check out the files and read the comments

## Changes
- `Api` and `History` api classes
- `useNetwork` hook
- `HistoryMockServer` driven by `msw`

## Checklist

### Testing

### Process

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #4486 

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
